scenarios/cpp/windows/captioning/caption.cpp

```diff
@@ -0,0 +1,580 @@
+/* NOTES:
+- In Windows, place the following files from the Speech SDK in the same folder as your compiled .exe:
+	- Microsoft.CognitiveServices.Speech.core.dll
+	- Microsoft.CognitiveServices.Speech.extension.audio.sys.dll
+*/
+
+/*
+Dependencies
+*/
+
+// exception
+#include <exception>
+// filesystem
+#include <filesystem>
+// ofstream
+#include <fstream>
+// fixed, setfill, setprecision, setw
+#include <iomanip>
+// cout
+#include <iostream>
+// optional
+#include <optional>
+// istringstream, ostringstream
+#include <sstream>
+
+/* Notes:
+- Install the NuGet package Microsoft.CognitiveServices.Speech.
+- The Speech SDK on Windows requires the Microsoft Visual C++ Redistributable for Visual Studio 2019 on the system. See:
+https://docs.microsoft.com/azure/cognitive-services/speech-service/speech-sdk?tabs=windows%2Cubuntu%2Cios-xcode%2Cmac-xcode%2Candroid-studio
+*/
+#include <speechapi_cxx.h>
+
+using namespace Microsoft::CognitiveServices::Speech;
+using namespace Microsoft::CognitiveServices::Speech::Speaker;
+
+/*
+Global constants
+*/
+
+// Note: Continuous language identification is supported only with v2 endpoints.
+const std::string v2EndpointStart ("wss://");
+const std::string v2EndpointEnd (".stt.speech.microsoft.com/speech/universal/v2");
+
+const float ticksPerSecond = 10000000.0;
+
+const std::string usage = "Usage: caption.exe [-d] [-f] [-h] [-i file] [-l] [-o file] [-p phrases] [-q] [-r number] [-s] [-t] [-u] <subscriptionKey> <region>\n"
+"              -d: Enable dictation mode.\n"
+"              -f: Enable profanity filter.\n"
+"              -h: Show this help and stop.\n"
+"              -i: Input audio file *file* (default input is from the microphone.)\n"
+"    -l languages: Enable language identification for specified *languages*.\n"
+"                  Example: en-US,ja-JP\n"
+"         -o file: Output to *file*.\n"
+"      -p phrases: Add specified *phrases*.\n"
+"                  Example: Constoso;Jessie;Rehaan\n"
+"              -q: Suppress console output (except errors).\n"
+"       -r number: Set stable partial result threshold to *number*.\n"
+"                  Example: 3\n"
+"              -s: Emit SRT (default is WebVTT.)\n"
+"              -t: Enable TrueText.\n"
+"              -u: Emit partial results instead of finalized results.\n";
+
+/*
+Types
+*/
+
+struct UserConfig
+{
+	const bool prioritizeAccuracyEnabled = false;
```

Comment:
We don't use TABS in our code, since it's unpredictable how that shows up in different editors. Replace every tab with 4 spaces. Set your editor to insert 4 spaces whenever you hit the tab key.